modules/kubernetes-addons/locals.tf

```diff
@@ -55,7 +55,7 @@ locals {
     aws_eks_cluster_endpoint       = local.eks_cluster_endpoint
     aws_partition_id               = data.aws_partition.current.partition
     aws_region_name                = data.aws_region.current.name
-    eks_cluster_id                 = var.eks_cluster_id
+    eks_cluster_id                 = data.aws_eks_cluster.eks_cluster.id
```

Comment:
this data source is tied to the `data_plane_wait_arn` which waits for the passed value to resolve before proceeding which helps to ensure the data plane is ready before the addons are provisioned